test-schema.json

```diff
@@ -27,6 +27,69 @@
+
+                "type": "array",
+                "minItems": 1,
+                "uniqueItems": true,
+                "items":{
+                    "properties": {
+                        "core": { 
+                            "description": "A section in official JSON Schema core drafts",
+                            "url": "https://json-schema.org/specification-links",
+                            "pattern": "^[0-9a-zA-Z]+(\\.[0-9a-zA-Z]+)*$", 
+                            "type":"string"
+                        },
+                        "validation": {
+                            "description": "A section in official JSON Schema validation drafts",
+                            "url": "https://json-schema.org/specification-links",
+                            "pattern": "^[0-9a-zA-Z]+(\\.[0-9a-zA-Z]+)*$", 
+                            "type":"string" 
+                        },
+                        "ecma262": { 
+                            "description": "A section in official ECMA 262 specification for defining regular expressions",
+                            "url": "https://262.ecma-international.org/",
+                            "pattern": "^[0-9a-zA-Z]+(\\.[0-9a-zA-Z]+)*$", 
+                            "type":"string" 
+                        },
+                        "perl5": { 
+                            "description": "A section name in Perl documentation for defining regular expressions",
+                            "url": "https://perldoc.perl.org/perlre",
+                            "type":"string" 
+                        },
+                        "quote": { 
+                            "description": "Quote describing the test case",
+                            "type":"string" 
+                        }
+                    },
+                    "patternProperties": {
+                        "^rfc\\d+$": { 
+                            "description": "A section in official RFC for the given rfc number",
+                            "url": "https://www.rfc-editor.org/",
+                            "pattern": "^[0-9a-zA-Z]+(\\.[0-9a-zA-Z]+)*$", 
+                            "type":"string" 
+                        },
+                        "^iso\\d+$": { 
+                            "description": "A section in official ISO for the given iso number",
+                            "pattern": "^[0-9a-zA-Z]+(\\.[0-9a-zA-Z]+)*$", 
+                            "type": "string" 
+                        }
+                    },
+                    "additionalProperties": { "type": "string" },
+                    "minProperties": 1,
+                    "propertyNames": {
+                        "oneOf": [
+                            {
+                                "pattern": "^((iso)|(rfc))[0-9]+$"
+                            },
+                            {
+                                "enum": [ "core", "validation", "ecma262", "perl5", "quote" ]
```

Comment:
Considering we need this enum and that IIRC `propertyNames` doesn't affect `unevaluatedProperties` (meaning we can't use that instead of the `additionalProperties` above) my suggestion (to use `propertyNames` was probably bad, but whatever, we can evolve the schema later if need be. Thanks for putting this in.